specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/preview/2021-09-01-preview/redhatopenshift.json

```diff
@@ -501,79 +554,122 @@
         }
       }
     },
+    "EncryptionAtHost": {
+      "description": "EncryptionAtHost represents encryption at host state",
+      "enum": [
+        "Disabled",
+        "Enabled"
+      ],
```

Comment:
Please check if you need x-ms-enum
https://github.com/stankovski/AutoRest/blob/master/Documentation/swagger-extensions.md#x-ms-enum

Reply:
We not added this to other "Two-state vars" Like, `Private`, `Public` or `Enable` `Disable`. I would like to keep this way, and if with the next version of API we will need, we can change  across the board.

Reply:
If you say we need to I can add, but I would prefer to add it to all String type switches (more change)

Reply:
`modelAsString` is actually your call, `false` or `true` is fine for SDK (though switch one to another going to be break changes).

The point is that you probably want to put a `name`. Else it could go unpredictable in SDK. This 2 `EncryptionAtHost` `EncryptionAtHost1` model in golang SDK might relate to the lack of `name`.
https://github.com/AzureSDKAutomation/azure-sdk-for-go/blob/078d46b49c7aaa98b99837e4755b4b9ecc40f51c/services/preview/mgmt/2021-09-01-preview/redhatopenshift/enums.go#L28-L56

Reply:
Will trust in your judgment here as Im not super familiar with all this area :)
Added for both (Visibility and EncryptionAtHost)

Reply:
Now it is only one of them https://github.com/AzureSDKAutomation/azure-sdk-for-go/blob/78a2b4b4486f787481f2fa9e51cd96356dbcea20/services/preview/mgmt/2021-09-01-preview/redhatopenshift/enums.go

I can merge when your service is ready.